packages/website/src/theme/BlogSidebar/Content/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { groupBlogSidebarItemsByYear } from '@docusaurus/plugin-content-blog/client';
+import { useThemeConfig } from '@docusaurus/theme-common';
+import type { Props } from '@theme/BlogSidebar/Content';
+import Heading from '@theme/Heading';
+import type { ReactNode } from 'react';
+import React, { memo } from 'react';
+import Markdown from 'react-markdown';
+
+import styles from './styles.module.css';
+
+function BlogSidebarYearGroup({
+  year,
+  yearGroupHeadingClassName,
+  children,
+}: {
+  year: string;
+  yearGroupHeadingClassName?: string;
+  children: ReactNode;
+}): React.JSX.Element {
+  return (
+    <div className={styles.blogSidebarContent} role="group">
+      <Heading as="h3" className={yearGroupHeadingClassName}>
+        {year}
+      </Heading>
+      {children}
+    </div>
+  );
+}
+
+function BlogSidebarContent({
+  items: itemsRaw,
+  yearGroupHeadingClassName,
+  ListComponent,
+}: Props): ReactNode {
+  const items = itemsRaw.map(item => ({
+    ...item,
+    // BlogSidebarItem only allows title: string, but we render as Markdown
+    title: (<Markdown>{item.title}</Markdown>) as unknown as string,
```

Comment:
2c: this is better as it will error for removal if/when the types are corrected

```suggestion
    // @ts-expect-error -- BlogSidebarItem only allows title: string, but we render as Markdown
    title: <Markdown>{item.title}</Markdown>,
```